configure.ac

```diff
@@ -370,18 +370,18 @@ esac
 if test "$enable_debug" = "yes"; then
   dnl If debugging is enabled, and the user hasn't overridden CXXFLAGS, clear
   dnl them, to prevent autoconfs "-g -O2" being added. Otherwise we'd end up
-  dnl with "-O0 -g3 -g -O2".
+  dnl with "-O0 -g4 -g -O2".
   if test "$CXXFLAGS_overridden" = "no"; then
   CXXFLAGS=""
   fi
 
   dnl Disable all optimizations
   AX_CHECK_COMPILE_FLAG([-O0], [DEBUG_CXXFLAGS="$DEBUG_CXXFLAGS -O0"], [], [$CXXFLAG_WERROR])
 
-  dnl Prefer -g3, fall back to -g if that is unavailable.
+  dnl Prefer -g4, fall back to -g if that is unavailable.
   AX_CHECK_COMPILE_FLAG(
-    [-g3],
-    [DEBUG_CXXFLAGS="$DEBUG_CXXFLAGS -g3"],
+    [-g4],
```

Comment:
I'm confused.

The `-g4` [sets](https://gcc.gnu.org/onlinedocs/gcc/Debugging-Options.html#Debugging-Options) debug information level 4, not DWARF version 4. And gcc-11 on Ubuntu 22.04 reports `g++: error: debug output level '4' is too high`.

Reply:
Yes this isn't correct, and should just be left as-is.